pkg/traceql/engine.go

```diff
@@ -40,17 +40,17 @@ func Compile(query string) (*RootExpr, SpansetFilterFunc, metricsFirstStageEleme
 
 	err = expr.validate()
 	if err != nil {
-		return nil, nil, nil, nil, err
+		return nil, nil, nil, nil, nil, err
 	}
 
-	return expr, expr.Pipeline.evaluate, expr.MetricsPipeline, req, nil
+	return expr, expr.Pipeline.evaluate, expr.MetricsPipeline, expr.MetricsSecondStage, req, nil
```

Comment:
As cumbersome as this is, I think it's the best way to start, since there is a lot of specialty in which pieces of the pipeline are executed by which modules.  The whole expression is never executed together.   Ideally we would create a holistic "metrics pipeline" that doesn't differentiate the stages so much, but it would be some work, and possibly introduce some wasted overhead (i.e. initializing span-level aggregation in the frontend that will never use it).